streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -318,6 +320,16 @@ public void foreach(ForeachAction<? super K, ? super V> action) {
     }
 
     @Override
+    public KStream<K, V> peek(final ForeachAction<? super K, ? super V> action) {
+        Objects.requireNonNull(action, "action can't be null");
+        final String name = topology.newName(PEEK_NAME);
+
+        topology.addProcessor(name, new KStreamPeek<>(action), this.name);
+
+        return new KStreamImpl<>(topology, name, sourceNodes, this.repartitionRequired);
```

Comment:
Wow, I really cause a lot of pain for myself by copying the `foreach` code's style ;)